haproxy/templates/haproxy.jinja

```diff
@@ -23,8 +23,14 @@
 # Global settings
 #------------------
 global
+{%- if salt['pillar.get']('haproxy:global:log', []) != [] %}
```

Comment:
You could just iterate over `salt['pillar.get']('haproxy:global:log', [])` directly here. If it is empty there is nothing to iterate over.